Remove jsonld framing in order to prevent duplicate data

elpoelma opened the pull request:
This replaces the framing of the json-ld results by a compaction. There was an issue with the framing of the data that resulted in duplicate data in the responses, this does not occur with compaction. In the future, we should look into framing the data without the problems with duplication.

elpoelma committed:
compact jsonld instead of framing in order to prevent duplicate data

aatauil commented:
I have been going down the rabbit hole with this :p 

- Currently I don't see the use of framing as we don't pass a frame object nor do we give the client the possibility to do so. (would be nice additions for later)

- What puzzles me is the output when compacting. Wiithout saving the initial @context given when the client POST's the data, compacting here just makes everything messy since it becomes a mix of compacted and expanded data. 

Currently when running jsonld.toRDF I noticed it outputs jsonld in an expanded format (which makes sense). Why not return that instead of semi-compacting the json-ld? That way the client can compact it themselfs with their own context. 

Just a suggestion from my perspective ofc, dont know what the spec or standards say about all this. Perhpas I am looking at the GET data too much as data they would use instead of data they look at to check if fragmented correctly

elpoelma committed:
remove jsonld compaction alltogether

elpoelma commented:
> I have been going down the rabbit hole with this :p
> 
>     * Currently I don't see the use of framing as we don't pass a frame object nor do we give the client the possibility to do so. (would be nice additions for later)
> 
>     * What puzzles me is the output when compacting. Wiithout saving the initial @context given when the client POST's the data, compacting here just makes everything messy since it becomes a mix of compacted and expanded data.
> 
> 
> Currently when running jsonld.toRDF I noticed it outputs jsonld in an expanded format (which makes sense). Why not return that instead of semi-compacting the json-ld? That way the client can compact it themselfs with their own context.
> 
> Just a suggestion from my perspective ofc, dont know what the spec or standards say about all this. Perhpas I am looking at the GET data too much as data they would use instead of data they look at to check if fragmented correctly

You're right, I just removed compaction alltogether and converted the turtle using the rdf-serializer library just like the other content-types.
